test/jdk/javax/swing/JTable/TestTableSelectAll.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4466930
+ * @key headful
+ * @summary Verifies selectAll selects all available rows and columns
+ *          irrespective of presence of columns and rows respectively
+ * @run main TestTableSelectAll
+ */
+
+import javax.swing.JTable;
+import javax.swing.table.DefaultTableModel;
+
+public class TestTableSelectAll {
+
+    public static void main(String[] args) throws Exception {
+
+        boolean colSelNoRow, colSelWithRow;
+
+        // TableModel with no rows, but 10 columns
+        DefaultTableModel data = new DefaultTableModel(0, 10);
```

Comment:
Proposed fix tries to handle two cases:
1. rowCount > 0 and columnCount = 0
2. columnCount > 0 and rowCount = 0

but test code covers only second case (rowCount = 0).

Test should be extended to verify both cases.

Reply:
Extended..